cpu/cc2538/cc2538.lds

```diff
@@ -103,6 +106,8 @@ SECTIONS
         *(.bss*)
         *(COMMON)
         _ebss = .;
+        //Begin of HEAP, END is absolute, see above.
+        _heapdata  = ABSOLUTE(.);
```

Comment:
This badly needs to be rebased because of the latest stack changes. Put the heap after the stack, filling the end of the SRAM. Or, the required heap size could be a configuration option given as an application-specific estimated value, in order to fail at build time rather than at run time if the heap is too small for the application requirements, but this is probably too much (it's up to the application developers to test the pointers returned by the allocation functions).

`_heap` rather than `_heapdata`.

Why `ABSOLUTE()`? Where are you using relocation?

In the end, the comments are no longer needed, and I would simplify the changes in this file like this:
```
     .stack (NOLOAD) :
     {
         *(.stack)
     } > SRAM
 
+    _heap = .;
+    _eheap = SRAM_START + SRAM_LEN;
+
 #if (LPM_CONF_MAX_PM==2) && (LPM_CONF_ENABLE != 0)
     .nrdata (NOLOAD) :
     {
         _nrdata = .;
```